core/src/main/java/org/apache/iceberg/hadoop/JdbcNamespace.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.hadoop;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import org.apache.iceberg.catalog.Namespace;
+
+public class JdbcNamespace {
+  private String catalogName;
+  private String namespace;
+  private String namespaceMetadata;
+
+  public JdbcNamespace(Namespace namespace) {
+    this.namespace = namespace.toString();
+  }
+
+  public String getCatalogName() {
+    return catalogName;
+  }
+
+  public void setCatalogName(String catalogName) {
+    this.catalogName = catalogName;
+  }
+
+  public String getNamespace() {
+    return namespace;
+  }
+
+  public void setNamespace(String namespace) {
+    this.namespace = namespace;
+  }
+
+  public String getNamespaceMetadata() {
+    return namespaceMetadata;
+  }
+
+  public void setNamespaceMetadata(String namespaceMetadata) {
+    this.namespaceMetadata = namespaceMetadata;
+  }
+
+  public Namespace toNamespace() {
+    return Namespace.of(namespaceMetadata);
+  }
+
+  public Map<String, String> toNamespaceMetadata() throws JsonProcessingException {
```

Comment:
For the initial version, let's not support namespace metadata. We can add that later in a separate PR where we consider more options for storage.

Reply:
relaying on `toString()` implementation didn't felt safe so for both `namespace metadata` and `Namespace` I used Json mapper to convert to string("json sting"), and from.

[JdbcUtil.java](https://github.com/ismailsimsek/iceberg/blob/iceberg_jdbccatalog/core/src/main/java/org/apache/iceberg/jdbc/JdbcUtil.java)

ex:
```
  public static String namespaceToString(Namespace namespace) throws JsonProcessingException {
    return JsonUtil.mapper().writer().writeValueAsString(namespace.levels());
  }
```